### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by franmomu**
| Q | A
| --- | ---
| Branch? | 3.x
| Bug fix? | no
| New feature? | no
| BC breaks? | yes
| Deprecations? | no
| Fixed tickets | #1406
| License | MIT
| Doc | 

Cleanup related to deprecated code

### CODE REVIEW

**Review by dbu — APPROVED** [review]
Awesome, so many things that can be deleted!

I fixed the UPGRADE.md in the 3.x branch - can you please rebase on 3.x and add the necessary information to the UPGRADE.md file for the BC breaks? i think we should keep that file up to date as we delete code. if we try to only do it later, we will forget some of the BC breaks.

File: composer.json [hunk 1]
```diff
@@ -39,7 +39,6 @@
         "aws/aws-sdk-php": "^2.4",
         "doctrine/cache": "^1.11|^2.0",
```
Comment (dbu) [inline composer.json]: do we still need this or can we get rid of it, now that CacheResolver has been removed?

**Review by dbu — APPROVED** [review]
thanks for the cleanup!

File: composer.json (hunk 1 above)
Comment (franmomu) [reply composer.json]: you're right, there are no references to doctrine cache, thanks!

**Comment by dbu** [comment]
thanks a lot!

**Pull request merged**